app/src/main/java/fr/free/nrw/commons/coordinates/CoordinateEditHelper.java

```diff
@@ -0,0 +1,182 @@
+                Latitude, Longitude, Accuracy, result)))
+            .firstOrError();
+    }
+
+    /**
+     * Replaces new coordinates
+     * @param media to be added
+     * @param Latitude to be added
+     * @param Longitude to be added
+     * @param Accuracy to be added
+     * @return Observable<Boolean>
+     */
+    private Observable<Boolean> addCoordinates(final Media media, final String Latitude,
+        final String Longitude, final String Accuracy) {
+        Timber.d("thread is coordinates adding %s", Thread.currentThread().getName());
+        final String summary = "Adding Coordinates";
+
+        final StringBuilder buffer = new StringBuilder();
+
+        final String wikiText = pageEditClient.getCurrentWikiText(media.getFilename())
+            .subscribeOn(Schedulers.io())
+            .blockingGet();
+
+        if (Latitude != null) {
+            buffer.append("\n{{Location|").append(Latitude).append("|").append(Longitude)
+                .append("|").append(Accuracy).append("}}");
+        }
+
+        final String editedLocation = buffer.toString();
+        final String appendText = getFormattedWikiText(wikiText, editedLocation);
+
+        return pageEditClient.edit(Objects.requireNonNull(media.getFilename())
+               , appendText, summary);
+    }
+
+    /**
+     * Helps to get formatted wikidata with upgraded location
+     * @param wikiText current wikitext
+     * @param editedLocation new location
+     * @return String
+     */
+    private String getFormattedWikiText(final String wikiText, final String editedLocation){
+
+        if(wikiText.contains("filedesc") && wikiText.contains("Location")) {
+
+            final String fromLocationToEnd = wikiText.substring(wikiText.indexOf("{{Location"));
+            final String firstHalf = wikiText.substring(0, wikiText.indexOf("{{Location"));
+            final String lastHalf = fromLocationToEnd.substring(
+                fromLocationToEnd.indexOf("}}") + 2);
+
+            return firstHalf + editedLocation + lastHalf;
+
+        }
+        if(wikiText.contains("filedesc") && !wikiText.contains("Location")){
+
+            final int startOfSecondSection = StringUtils.ordinalIndexOf(wikiText,
+                "==", 3);
+
+            if(startOfSecondSection != -1) {
+                final String firstHalf = wikiText.substring(0, startOfSecondSection);
+                final String lastHalf = wikiText.substring(startOfSecondSection);
+                return firstHalf + editedLocation + lastHalf;
+            }
+            return wikiText + editedLocation;
+        }
+        if(!wikiText.contains("filedesc") && !wikiText.contains("Location")){
+
+            return "== {{int:filedesc}} ==" + editedLocation+wikiText;
+
+        }
+        if(!wikiText.contains("filedesc") && wikiText.contains("Location")){
+
+            return "== {{int:filedesc}} ==" + editedLocation+wikiText;
```

Comment:
Sorry I had missed this one: spaces before/after `+`

Reply:
Sorry, I overlooked this one.